File: pset6/caesar/caesar.py
"""
/**
 * J.T. Williams
 *
 * Tech901 Code1.0
 *
 * CS50
 *
 * Problem Set 6
 *
 * caesar.py
 *
 * Executes caesar.c as caesar.py
 */
"""

import cs50
import sys

# starting python code mirror to c code


def main():
    if len(sys.argv) != 2:
        sys.exit(1)
    key = int(sys.argv[1])
    print("plaintext: ", end="")
    plaintext = cs50.get_string()
    print()
    print("ciphertext: ", end="")
    for character in plaintext:
        if str.isalpha(character):
            if str.isupper(character):
                print(chr(((ord(character) + key - 65) % 26) + 65), end="")
            else:
                print(chr(((ord(character) + key - 97) % 26) + 97), end="")
        else:
            print(character, end="")
    print()


main()

File: pset2/caesar/caesar.c
/****************************
 * JT Williams              *
 * Tech901                  *
 * CS50 Problem Set 2       *
 * caesar.c                 *
 ****************************/

 #include <stdio.h>
 #include <cs50.h>
 #include <ctype.h>
 #include <string.h>
 #include <stdlib.h>


//start a main block for command line input
 int main(int argc, string argv[])
 {

  if(argc != 2)
  {

   printf("ERROR\n");
   return 1;

  }

   //declare ks as a string of argv[1]
   string ks = argv[1];

   //convert ki to an integer
   int ki = atoi(ks);

   printf("plaintext: ");

   //declaring & prompting the user for plaintext
   string plaintext = get_string();

   //print ciphertext
   printf("ciphertext: ");

   for (int i = 0, n = strlen(plaintext); i < n; i++)
   {

    //check if the char is alphabetic
    if (isalpha(plaintext[i]))
    {

     //preserve the case of the char
     if (isupper(plaintext[i]))
     {
      //print the capital char by the key
      printf("%c", ((plaintext[i] + ki - 65) % 26) + 65);

     }
     else
     {

      //print the lower case char by the key
      printf("%c", ((plaintext[i] + ki - 97) % 26) + 97);

     }

    }
    else
    {

     //if the char for plaintext[i] is not alphabetic, return the char
     printf("%c", plaintext[i]);

    }

   }

   //print new line
   printf("\n");

 }
